opentelemetry/src/metrics/descriptor.rs

```diff
@@ -73,6 +74,11 @@ impl Descriptor {
         self.config.unit.as_ref().map(|unit| unit.as_ref())
     }
 
+    /// Assign a new unit.
+    pub fn set_unit(&mut self, unit: Unit) {
```

Comment:
Hm we probably don't want this in the public api. Maybe marking the descriptor properties as `pub(crate)` would be simpler?

Reply:
I removed this method and made the `config` property `pub(crate)` to access it directly in builders.